test/unit/puppetlabs/puppetserver/certificate_authority_test.clj

```diff
@@ -154,6 +154,26 @@
                     :actual actual#})
         actual#))))
 
+;; This additional slingshot helper assumes our :kind/:msg maps and is probably
+;; brittle.
+(defmethod assert-expr 'thrown-with-slingshot-msg? [msg form]
+  (let [expected (nth form 1)
+        body     (nthnext form 2)]
+    `(sling/try+
+      ~@body
+      (do-report {:type :fail :message ~msg :expected ~expected :actual nil})
+      (catch map? actual#
+        (let [result# (if (or (= actual# ~expected)
+                              (and (= (:kind actual#) (:kind ~expected))
+                                   (re-matches (:msg ~expected) (:msg actual#))))
+                        :pass
+                        :fail)]
+          (do-report {:type result#
+                      :message ~msg
+                      :expected ~expected
+                      :actual actual#}))
+        actual#))))
+
```

Comment:
If memory serves, `thrown-with-slingshot?` exists because, at the time, we either didn't know about `slingshot.test/thrown+?` or couldn't get it to work.  Since then, I've used it elsewhere successfully.

There's also a `thrown+-with-msg?` in that same namespace, which looks very similar to this with the exception that it doesn't know about our error format.  

Given that, does it make sense to add this?